### PR TITLE
Use CancelIoEx to notify input thread

### PR DESCRIPTION
**Pull request by tigerw**
- Removed potentially fragile calls to send a virtual keypress to the console.

### CODE REVIEW

File: src/Globals.h [hunk 1]
```diff
@@ -187,7 +187,7 @@ template class SizeChecker<UInt8,  1>;
 #ifdef _WIN32
 
 	#define WIN32_LEAN_AND_MEAN
-	#define _WIN32_WINNT _WIN32_WINNT_WS03  // We want to target Windows XP with Service Pack 2 & Windows Server 2003 with Service Pack 1 and higher
```
Comment (bearbin) [inline src/Globals.h]: This comment seems outdated.

File: src/Root.cpp [hunk 2]
```diff
@@ -252,7 +252,7 @@ void cRoot::Start(std::unique_ptr<cSettingsRepositoryInterface> a_OverridesRepo)
 
 		LOG("Shutting down server...");
 		m_Server->Shutdown();
-	}  // if (m_Server->Start()
```
Comment (bearbin) [inline src/Root.cpp]: Why was this comment removed?

Comment (tigerw) [reply src/Root.cpp]: It didn't seem particularly useful to me :P

**Comment by tigerw** [comment]
Also, does anyone know why pthread_kill returns error 3, yet still appears to work and stop the thread? It's giving a false error every time the server stops on Linux.

File: src/Root.cpp (hunk 2 above)
Comment (bearbin) [reply src/Root.cpp]: I think it's probably better left - it just gives an indication of what the long indented block was for.

**Comment by bearbin** [comment]
Error 3 suggests that the thread being killed is already dead. I have no idea why it would happen though.

**Comment by madmaxoft** [comment]
Ugh, again my review got stuck and wasn't posted.

Do we really want to lose XP compatibility over something that works anyway? I don't think this PR is necessary and it won't be needed in the long run (once Cuberite is split into the library part and the (G)UI part).

**Comment by tigerw** [comment]
Is anyone running the server on XP? It seems unlikely.

When is this split projected to happen?

**Comment by bearbin** [comment]
We shouldn't reduce compatibility just for the sake of reducing compatibility. You can run Cuberite on an Android device, even though you would be mad to do so - why not XP?

Also interested to know about this proposed split - can you link me some more information?

**Comment by madmaxoft** [comment]
Here's one forum thread about the split, though I seem to recall it being discussed in other places, too, but can't find it.
https://forum.cuberite.org/thread-2740.html

Quite a lot of hardware is limited to XP (or equivalent), so let's just not cut it away.

[OT] Man, it would be awesome to actually see Cuberite running on an ATM :D

**Comment by tigerw** [comment]
But sending virtual keypresses to a console window to close it is hardly a permanent solution.

Why are we so hung up about Windows XP?
________________________________

**Comment by madmaxoft** [comment]
Yes, it's not a permanent solution, it's there just until the library / UI split.

Incidentally, did you test your change with the `restart` console command and webadmin button? Does it work, and does it let the console to receive further commands after the restart?

**Comment by peterbell10** [comment]
Closing as this has been inactive for over 6 months.

**Comment by tigerw** [comment]
Well, whatever. @madmaxoft I do believe restart worked correctly, but I may have to try again to confirm.

**Pull request closed without merging**